src/modules/AutoinstConfig.rb

```diff
@@ -522,6 +527,38 @@ def MainHelp
       main_help
     end
 
+    # Reports name of base product as obtained from AY profile
+    #
+    # There are several ways how can base product be defined in the profile
+    # 1) explicitly
+    # 2) impllicitly according software selection
+    #
+    # @return [Y2Product] a base product or nil
+    def selected_product
+      return @selected_product if @selected_product
+
+      # magic for selecting base product according to the profile
+      # check if base product is defined explicitly in the profile
+      software = Profile.current.fetch("software", {})
+      base_products = Y2Packager::Product.available_base_products
+
+      # try to find base product for installation according the profile
+      product_name = software.fetch("products", {})["product"]
+      return base_products.find { |p| p.short_name == product_name } if product_name
+
+      # try to find product according to patterns in profile
+      # searching for patterns like "sles-base-32bit"
+      products = base_products.select do |product|
```

Comment:
These 4 lines must go into a separate method, because they are a complex piece of logic (and the source of that "magic" comment above, I believe).
Furthermore, I would like this method not to exist at all. Was it really requested? Why not simply use the explicit product requested in the profile?

Reply:
explained offline.
refactoring is wip